sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -872,6 +886,25 @@ object PushDownPredicate extends Rule[LogicalPlan] with PredicateHelper {
       pushDownPredicate(filter, u.child) { predicate =>
         u.withNewChildren(Seq(Filter(predicate, u.child)))
       }
+
+    case filter @ Filter(condition, watermark: EventTimeWatermark) =>
```

Comment:
I see. How about moving this case above `UnaryNode`? The following two functions `canPushThrough ` and `pushDownPredicate ` are just for UnaryNode.